cylc/flow/task_pool.py

```diff
@@ -697,46 +704,99 @@ def rh_release_and_queue(self, itask) -> None:
             self.queue_task(itask)
 
     def _get_spawned_or_merged_task(
-        self, point: 'PointBase', name: str, flow_nums: 'FlowNums'
-    ) -> Optional[TaskProxy]:
-        """Return new or existing task point/name with merged flow_nums"""
-        taskid = Tokens(cycle=str(point), task=name).relative_id
+        self, point: 'PointBase', tdef: 'TaskDef', flow_nums: 'FlowNums'
+    ) -> 'Tuple[Optional[TaskProxy], bool, bool]':
+        """Return new or existing task point/name with merged flow_nums
+
+        Returns:
+            tuple - (itask, is_in_pool, is_xtrig_sequential)
+
+            itask:
+                The requested task proxy, or None if task does not
+                exist or cannot spawn.
+            is_in_pool:
+                Was the task found in a pool.
+            is_xtrig_sequential:
+                Is the next task occurance spawned on xtrigger satisfaction,
+                or do all occurances spawn out to the runahead limit.
+
+        """
+        taskid = Tokens(cycle=str(point), task=tdef.name).relative_id
         ntask = (
             self._get_hidden_task_by_id(taskid)
             or self._get_main_task_by_id(taskid)
         )
+        is_in_pool = False
+        is_xtrig_sequential = False
         if ntask is None:
             # ntask does not exist: spawn it in the flow.
-            ntask = self.spawn_task(name, point, flow_nums)
+            ntask = self.spawn_task(tdef.name, point, flow_nums)
+            # if the task was found set xtrigger spawning type.
+            # otherwise find the xtrigger type if it can't spawn
+            # for whatever reason.
+            if ntask is not None:
+                is_xtrig_sequential = ntask.is_xtrigger_sequential
+            elif any(
+                xtrig_label in self.xtrigger_mgr.sequential_xtrigger_labels
+                for sequence, xtrig_labels in tdef.xtrig_labels.items()
+                for xtrig_label in xtrig_labels
+                if sequence.is_valid(point)
+            ):
+                is_xtrig_sequential = True
         else:
             # ntask already exists (n=0 or incomplete): merge flows.
+            is_in_pool = True
             self.merge_flows(ntask, flow_nums)
-        return ntask  # may be None
+            is_xtrig_sequential = ntask.is_xtrigger_sequential
+        # ntask may still be None
+        return ntask, is_in_pool, is_xtrig_sequential
 
     def spawn_to_rh_limit(self, tdef, point, flow_nums) -> None:
-        """Spawn parentless task instances from point to runahead limit."""
+        """Spawn parentless task instances from point to runahead limit.
+
+        Sequentially checked xtriggers will spawn the next occurrence of their
+        corresponding tasks. These tasks will keep spawning until they depend
+        on any unsatisfied xtrigger of the same sequential behavior, are no
+        longer parentless, and/or hit the runahead limit.
+
```

Comment:
```suggestion
```

Isn't this describing the spawning of tasks that don't go via this method? (So the comment should really be elsewhere).

Reply:
Well, I added this as a caveat to `spawn_to_rh_limit` .. Because as described, it won't always spawn to the RH limit.

Reply:
Oh sorry, misread, this method _is_ used to spawn x-sequentials, it just bails out early.